### PR TITLE
add preview builder

### LINKED ISSUE #11: Custom content in ContextMenu
Simulator Screen Shot - iPhone 11 - 2020-07-29 at 16 45 11

After showing the context menu, I just want to display the image of the item.
How to customize content after showing context menu?

### PR DESCRIPTION
**Pull request by nucleartux**
Hello,

I added possibility to specify preview builder.

Here is a example:
```
<ContextMenu>
        <View style={[styles.rectangle, {backgroundColor: 'red'}]} />
        <View style={[styles.rectangle, {backgroundColor: 'green'}]} />
</ContextMenu>
```

First child is a view displayed on screen and second is a preview.

I think it's breaking change because now you can only add single child inside component.

So any suggestions would be greatly appreciated.

Closes #11.

### CODE REVIEW

**Comment by mpiannucci** [comment]
Awesome thanks for this. I think a better API for this is to pass the preview View as a prop instead of as a second child, leaving open the possibility to wrap multiple views in one context view in the future. What do you think about that? It would be like: 

```jsx
<ContextMenu preview={<View style={[styles.rectangle, {backgroundColor: 'green'}]} />}>
        <View style={[styles.rectangle, {backgroundColor: 'red'}]} />
</ContextMenu>
```

**Comment by nucleartux** [comment]
@mpiannucci Done

**Review by mpiannucci — APPROVED** [review]
This is so great. LGTM

File: index.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+import React from "react";
+import { requireNativeComponent, View } from "react-native";
 
-const ContextMenu = requireNativeComponent('ContextMenu', null);
+const NativeContextMenu = requireNativeComponent("ContextMenu", null);
+
+const ContextMenu = (props) => {
+  return (
+    <NativeContextMenu {...props}>
+      <View>{props.children}</View>
```
Comment (jacobp100) [inline index.js:9]: What happens when view flattening is turned on for iOS?

I’d be tempted to flip this - don’t wrap the main children, if there’s a preview, wrap that but set the [nativeID](https://reactnative.dev/docs/view#nativeid) prop, which will disable view flattening, and you can [read in obj c](https://github.com/facebook/react-native/blob/7b05b091fd97f95b778369277ac2147730abc7b8/React/Views/UIView%2BReact.m#L38) to check if it’s the preview

**Comment by jacobp100** [comment]
Great work! So glad to see this done. Just had one thought to future proof this against the new architecture

File: index.js (hunk 1 above)
Comment (mpiannucci) [reply index.js:9]: Hmm you got a link about the view flattening? Haven't read about that yet

**Comment by nucleartux** [comment]
@jacobp100 can you implement this please? To be honest i'm not big expert in ios.

Also I noticed bug with preview: it doesn't appear on screen but it takes space. Maybe you know how to fix this behaviour?

File: index.js (hunk 1 above)
Comment (jacobp100) [reply index.js:9]: https://reactnative.dev/architecture/view-flattening

In short, it won't produce UIViews for elements that are only there for layout - which is the case here

**Comment by jacobp100** [comment]
I'll try and find some time

**Pull request merged**